zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/callback/ZkAsyncCallbacks.java

```diff
@@ -289,7 +290,11 @@ protected boolean needRetry(int rc) {
           return false;
         }
       } catch (ClassCastException | NullPointerException ex) {
-        LOG.error("Failed to handle unknown return code {}. Skip retrying.", rc, ex);
+        if(rc == ZK_SESSION_MISMATCHED_CODE) {
```

Comment:
This should not fall into Exception handling logic, it is the expected RC number.
So if we are moving all the return code into a new class, then the switch case should check using that new class, and this is no longer a special case.
If, otherwise, we are not using a new class then, we can check this condition before the switch block.

Reply:
Thanks for the comment, I moved it before the switch block. For the moment I haven't defined a new class, but if I do so I will use that class in the switch block.